components/kyma-environment-broker/internal/storage/postsql/read.go

```diff
@@ -832,12 +832,16 @@ func addOrchestrationFilters(stmt *dbr.SelectStmt, filter dbmodel.OrchestrationF
 
 func addOperationFilters(stmt *dbr.SelectStmt, filter dbmodel.OperationFilter) {
 	if len(filter.States) > 0 {
-		stmt.Where("state IN ?", filter.States)
+		stmt.Where("o.state IN ?", filter.States)
 	}
 	if filter.InstanceFilter != nil {
 		fi := filter.InstanceFilter
+		if fi.OnlyDeleted != nil && *fi.OnlyDeleted {
+			stmt.LeftJoin(dbr.I(InstancesTableName).As("i"), "i.instance_id = o.instance_id").
```

Comment:
1) do we have tests for checking those functionalities?
2) call like: 
```kcp rt -i instance123 --only-deleted --events``` will return events only for instances that were deleted, right?

Reply:
> 1. do we have tests for checking those functionalities?

we don't have any tests in the `postsql` package, there was a plan to introduce some in-memory db for unit tests so these queries can be tested, but not sure how far the topic got.

> 2. call like:
`kcp rt -i instance123 --only-deleted --events` will return events only for instances that were deleted, right?

correct, if `instance123` is deleted, it will display events for that instance. If the instance isn't deleted (there is still an entry in the `instances` table